tsl/src/dist_util.c

```diff
@@ -113,6 +127,33 @@ dist_util_set_id(Datum dist_id)
 					 (errmsg("database is already a member of a distributed database"))));
 	}
 
+	if (check_metadata)
+	{
+		uuid = ts_metadata_get_value(CStringGetDatum(METADATA_UUID_KEY_NAME),
+									 CSTRINGOID,
+									 UUIDOID,
+									 &isnull);
+		if (isnull)
+			ereport(ERROR,
+					(errcode(ERRCODE_TS_DATA_NODE_INVALID_CONFIG),
+					 (errmsg("uuid not set for data node"))));
+		else if (uuid_matches(uuid, dist_id))
+		{
+			Datum dist_uuid_str = DirectFunctionCall1(uuid_out, dist_id);
+
+			ereport(ERROR,
+					(errcode(ERRCODE_TS_DATA_NODE_SELF_REFERENCE),
+					 (errmsg("attempting to add self as data node"),
+					  errdetail("Trying to set dist_uuid to %s, but uuid is the same, so this "
+								"would create a cycle.",
+								DatumGetCString(dist_uuid_str)),
+					  errhint(
+						  "Do not try to add the database you're connected to as data node. Either "
+						  "use a different database on the same instance, or a different "
+						  "instance."))));
```

Comment:
The first part of this error is self-evident, so I suggest to elide. Also, second part is misleading because it sounds like you need to use an existing database. I don't recall if we support already existing databases or whether we require creating one. In any case, I suggest something like:

```suggestion
					  errhint(
						  "Use a database name that isn't already part of a distributed database."))));
```